src/deployment/deploylib/configuration.py

```diff
@@ -0,0 +1,228 @@
+#!/usr/bin/env python
+#
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+import ipaddress
+import json
+import logging
+from typing import Any, Dict, List, Optional
+from uuid import UUID
+
+from azure.cosmosdb.table.tableservice import TableService
+
+storage_client_logger = logging.getLogger("azure.cosmosdb.table.common.storageclient")
+TABLE_NAME = "InstanceConfig"
+
+logger = logging.getLogger("deploy")
+
+
+class InstanceConfigClient:
+    table_service: TableService
+    resource_group: str
+
+    def __init__(self, table_service: TableService, resource_group: str):
+        self.resource_group = resource_group
+        self.table_service = table_service
+        self.create_if_missing(table_service)
+
+    ## Disable logging from storageclient. This module displays an error message
+    ## when a resource is not found even if the exception is raised and handled internally.
+    ## This happen when a table does not exist. An error message is displayed but the exception is
+    ## handled by the library.
+    def disable_storage_client_logging(self) -> None:
+        if storage_client_logger:
+            storage_client_logger.disabled = True
+
+    def enable_storage_client_logging(self) -> None:
+        if storage_client_logger:
+            storage_client_logger.disabled = False
+
+    def create_if_missing(self, table_service: TableService) -> None:
+        try:
+            self.disable_storage_client_logging()
+
+            if not table_service.exists(TABLE_NAME):
+                table_service.create_table(TABLE_NAME)
+        finally:
+            self.enable_storage_client_logging()
+
+
+class NetworkSecurityConfig:
+    allowed_ips: List[str]
+    allowed_service_tags: List[str]
+
+    def __init__(self, config: Any):
+        self.parse_nsg_json(config)
+
+    def parse_nsg_json(self, config: Any) -> None:
+        if not isinstance(config, Dict):
+            raise Exception(
+                "Configuration is not a Dictionary. Please provide Valid Config."
+            )
+        if len(config.keys()) == 0:
+            raise Exception(
+                "Empty Configuration File Provided. Please Provide Valid Config."
+            )
+        if None in config.keys() or "proxy_nsg_config" not in config.keys():
+            raise Exception(
+                "proxy_nsg_config not provided as valid key. Please Provide Valid Config."
+            )
+        proxy_config = config["proxy_nsg_config"]
+        if not isinstance(proxy_config, Dict):
+            raise Exception(
+                "Inner Configuration is not a Dictionary. Please provide Valid Config."
+            )
+        if len(proxy_config.keys()) == 0:
+            raise Exception(
+                "Empty Inner Configuration File Provided. Please Provide Valid Config."
+            )
+        if (
+            None in proxy_config.keys()
```

Comment:
What is the purpose of these `None in ..` checks? cc @nharper285?

Reply:
I'll fix it